src/services/createPropStore.ts

```diff
@@ -111,6 +114,59 @@ export function createPropStore(): PropStore {
     return store.get(key)
   }
 
+  function getParentContext(route: ResolvedRoute, prefetch: boolean = false): PropsCallbackParent {
+    const parent = route.matches.at(-2)
+
+    if (!parent) {
+      return
+    }
+
+    if (isWithComponentProps(parent)) {
+      return {
+        name: parent.name ?? '',
+        get props() {
+          return getParentProps(parent, 'default', route, prefetch)
+        },
+      } as PropsCallbackParent
```

Comment:
@stackoverfloweth I was struggling to get this file to be happy with the types. All the api types and functionality are tested. So I'm okay with some hacks in here. But be nice if they were not necessary. Maybe you can see a nicer solution.

Reply:
actually I think we can pivot your `GetParentPropsReturnType` slightly and it all just works

Reply:
I'm slightly worried that'll affect the types you get in createRoute. I'll have to write a type test to make sure it says props is undefined when there is a parent but there are no props.

Reply:
@stackoverfloweth I added a type test for this that is failing. If we change `GetParentPropsReturnType` to return `unknown` then when a parent has no props the child thinks they are `unknown` rather than `undefined.